src/components/forms/AddressInput/index.tsx

```diff
@@ -38,51 +50,88 @@ const AddressInput = ({
   inputAdornment,
   validators = [],
   defaultValue,
+  value = '',
   disabled,
-}: AddressInputProps): React.ReactElement => (
-  <>
-    <Field
-      className={className}
-      component={TextField as any}
-      defaultValue={defaultValue}
-      disabled={disabled}
-      inputAdornment={inputAdornment}
-      name={name}
-      placeholder={placeholder}
-      testId={testId}
-      text={text}
-      type="text"
-      spellCheck={false}
-      validate={composeValidators(required, mustBeEthereumAddress, ...validators)}
-    />
-    <OnChange name={name}>
-      {async (value: string) => {
-        const address = trimSpaces(value)
-        // A crypto domain name
-        if (isValidEnsName(address) || isValidCryptoDomainName(address)) {
-          try {
-            const resolverAddr = await getAddressFromDomain(address)
-            const formattedAddress = checksumAddress(resolverAddr)
-            fieldMutator(formattedAddress)
-          } catch (err) {
-            logError(Errors._101, err.message)
-          }
-        } else {
-          // A regular address hash
-          let checkedAddress = address
-          // Automatically checksum valid (either already checksummed, or lowercase addresses)
-          if (isValidAddress(address)) {
+}: AddressInputProps): React.ReactElement => {
+  const showNetworkPrefix = useSelector(showShortNameSelector)
+  const [prefixedAddress, setPrefixedAddress] = React.useState(() => addNetworkPrefix(value, showNetworkPrefix))
+  const [prefixedError, setPrefixedError] = React.useState(false)
+
+  return (
+    <>
+      <Field
+        className={className}
+        component={TextField as any}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        inputAdornment={inputAdornment}
+        name={name}
+        inputProps={{
+          value: showNetworkPrefix ? prefixedAddress : value,
+          onChange: (e) => {
```

Comment:
You are using two `onChange` events in this component. Can they be combined in one?

Reply:
The onChange of the line 71 is related with the input, which contains the prefixed address value (`rin:0x123...` format).
The other onChange is related with the value that is stored in the react-final-form state (address without prefix `0x123...` format).
It can not be combined because if we change the address formats in the state of all forms, we will need to update all the flows to adapt them to the new prefixed address format.

Reply:
As per our call, you could theoretically remove the input `onChange` and also reliance on the state by prepending the `shortName` here:

```ts
inputProps={{
    value: `${showPrefix ? getCurrentShortChainName() : ''}${values[name]}`
}}
```